common/admin/smb/cluster.go

```diff
@@ -55,6 +63,41 @@ type PublicAddress struct {
 	Destination []string
 }
 
+// CustomPortsMap is used to configure a cluster with custom ports for
+// a specified service type.
+type CustomPortsMap map[Service]int
```

Comment:
Should we add a validation for any unknown `Service` set within _CustomPortsMap_?

Reply:
I think I'd rather skip that. My reason is that if people actually reach for `Service("some-other-name")` rather than using one of the actual enumerated values then they're probably doing so intentionally. For example, if go-ceph is lagging behind on the list of services the mgr side currently supports. Since the server *will* also validate the input values here and reject genuinely unknown service names I think it's extra code and it cuts off a possible 'escape hatch'.

Make sense?